controllers/argocd/applicationset_test.go

```diff
@@ -77,7 +77,12 @@ func TestReconcileApplicationSet_CreateDeployments(t *testing.T) {
 	a := makeTestArgoCD()
 	a.Spec.ApplicationSet = &argoproj.ArgoCDApplicationSet{}
 
-	r := makeTestReconciler(t, a)
+	resObjs := []client.Object{a}
+	subresObjs := []client.Object{a}
+	runtimeObjs := []runtime.Object{}
+	sch := makeTestReconcilerScheme(argoproj.AddToScheme)
+	cl := makeTestReconcilerClient(sch, resObjs, subresObjs, runtimeObjs)
+	r := makeTestReconciler(cl, sch)
```

Comment:
Is there a reason these changes are not done inside of `makeTestReconciler` function but in this way?

Reply:
Yes, its more modular to have the scheme and client defined within the consuming test because each test can define what kind of objects it needs to pass, as well as what groups need to be registered with the scheme 

some tests need to register resources with status subresource before the test begins, while others don't
this is a change that needs to be accommodated due to the breaking changes in controller runtime after the upgrade to `v0.16.3`, so we can't have a one-size fits all solution inside `makeTestReconciler` anymore